public/app/core/components/SVG/utils.ts

```diff
@@ -0,0 +1,27 @@
+import { v4 as uuidv4 } from 'uuid';
+
+export const getSvgStyle = (svgCode: string) => {
+  const svgStyle = svgCode.match(new RegExp('<style type="text/css">([\\s\\S]*?)<\\/style>'));
+  return svgStyle ? svgStyle[0] : null;
+};
+
+export const getSvgId = (svgCode: string) => {
+  return svgCode.match(new RegExp('<svg.*id\\s*=\\s*([\'"])(.*?)\\1'))?.[2];
+};
+
+export const svgStyleCleanup = (svgCode: string) => {
+  let svgId = getSvgId(svgCode);
+  if (!svgId) {
+    svgId = `x${uuidv4()}`;
+    const pos = svgCode.indexOf('<svg') + 5;
```

Comment:
any reason in particular for +5? We might want to consider replacing magic number with variable (something like `POSITION_OFFSET`?)